File: JavaTraining/src/main/java/Selenium/Common/PageEntities/PageBase.java
package Selenium.Common.PageEntities;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.openqa.selenium.WebDriver;

public class PageBase {
    private Logger log = LogManager.getFormatterLogger(this.getClass().getName());
    protected WebDriver webDriver;
    public PageBase parent;

    protected PageBase() {}

    public PageBase(WebDriver webDriver, PageBase parent) {
        this.webDriver = webDriver;
        this.parent = parent;
    }
}

File: C#/SeleniumTest/SeleniumTest/Core/PageEntities/PageBase.cs
using log4net;
using OpenQA.Selenium;

namespace SeleniumTest.Core
{
    public class PageBase
    {
        private ILog log;
        protected IWebDriver webDriver;
        public PageBase parent;

        protected PageBase() { }

        public PageBase(IWebDriver webDriver, PageBase parent)
        {
            this.webDriver = webDriver;
            this.parent = parent;

            log = LogManager.GetLogger(this.GetType());
        }
    }
}
